bionic/flow.py

```diff
@@ -1557,6 +1557,19 @@ def ...
 
         return flow
 
+    def reloading_in_place(self):
+        """
+        Attempts to reload all modules used directly by this flow and updates this flow instance.
+
+        Note that unlike other operations which are immutable, this method mutates the flow instance.
+
+        Please see ``Flow.reloading()`` for caveats.
+        """
+        flow = self.reloading()
+        self._state = flow._state
+        self._deriver = flow._deriver
+        self._uuid = flow._uuid
```

Comment:
Our documentation says that we compute an entity only once for a flow instance when it is decorated with `@changes_per_run`. Since we are not changing the instance and just mutating the current flow, I think we should preserve the old UUID and not compute it again.

I think testing it with `@changes_per_run` decorator is a good idea.

Reply:
Added `@changes_per_run` tests

Reply:
Thinking about this more: although we are technically keeping the "same" flow instance, it's not obvious to me that we want to keep the same UUID. I believe most users will think of `@changes_per_run` as "recompute every time", but we have implemented it as "recompute once for each flow instance" because we want to keep each flow instance's in-memory cache consistent. However, in this case we are clearing the in-memory cache (by resetting `_deriver`) so this reason doesn't apply. I think resetting the UUID and allowing all the `@changes_per_run` entities to be recomputed may be a more intuitive behavior.

@namanjain WDYT?

Reply:
The way I think about it, users are probably used to the fact that unless the flow changes, changes_per_run entities are not recomputed. So I actually think that it would be confusing if we recompute them if the flow variable still points to the same object.

But now that I said it out loud, it would also be confusing that we'll recompute those entities if the input changes with reload (because the provenance will change). But we won't recompute them if the input doesn't change. To have consistent behavior, I'm now convinced that we should recompute them. 

I think it would be good to update changes per run documentation to reflect this. Wdyt?

Reply:
Copying my comment from Slack: since this is an interaction between `changes_per_run` and `reload`, and the latter is the weird special case, I think we should document it in `reload`.

Reply:
After our conversation, updated the method to use a new UUID. Entities marked with `changes_per_run` are recomputed.